react/ShelfContent.js

```diff
@@ -94,6 +96,19 @@ class ShelfContent extends Component {
     )
   }
 
+  pushPixelProductsImpressions = () => {
+    const { products } = this.props
+    if (products)
+      products.forEach((product, index) => {
+        this.props.push({
+          event: 'productImpression',
+          list: 'Shelf',
+          position: index + 1,
+          product,
```

Comment:
Create some way to send the same product of the other events, example of missing property `selectedSku`.